src/middleware/authenticateToken.ts

```diff
@@ -0,0 +1,58 @@
+// authenticateToken.ts
+
+import jwt from 'jsonwebtoken';
+import { Request, Response, NextFunction } from 'express';
+import properties from '../properties';
+import { AuthenticatedUser } from '../types/AuthenticatedUser';
+import logger from '../logger';
+
+const SECRET_KEY = properties.secretKey;
+
+export function authenticateToken(req: Request, res: Response, next: NextFunction) {
+
+  try {
+    const token = req.headers['authorization'];
+    if (typeof token === 'string') {
+      const tokenString = token.split(' ')[1]; 
+      const user = jwt.verify(tokenString, SECRET_KEY);
+      req.user = user; 
+      next();
+    } else {
+      return res.status(401).json({
+        success: false,
+        error: {
+          error: true,
+          message: 'Unauthorized',
+          path: req.baseUrl,
+        },
+      })
+    } 
+  } catch (err) {
+    logger.error(`Error occured in authenticateToken : ${err}`);
+    return res.status(403).json({
+      success: false,
+      error: {
+        error: true,
+        message: String(err),
+        path: req.baseUrl,
+      },
+    })
+  }
+}
+
+export function checkAuthorization(req: Request, res: Response, next: NextFunction) {
+  const tokenUserId = Number((req.user as AuthenticatedUser).user_id);
+  const requestedUserId = Number(req.params.id); 
+  if (tokenUserId !== requestedUserId) {
+    return res.status(403).json({
+      success: false,
+      error: {
+        error: true,
+        message: 'Users are forbidden to update other users',
```

Comment:
message should be more generic. We could be simply say `Permission Denied. Forbidden.`

Reply:
Added Error message in constants and handled common error message.